modules/evm-manager/src/lib.rs

```diff
@@ -87,44 +90,109 @@ impl<T: Config> CurrencyIdMapping for EvmCurrencyIdMapping<T> {
 	// Take the first 4 non-zero bytes, if it is less than 4, add 0 to the left.
 	#[require_transactional]
 	fn set_erc20_mapping(address: EvmAddress) -> DispatchResult {
-		let id: u32 = CurrencyId::Erc20(address)
-			.try_into()
-			.expect("CurrencyId::Erc20 into u32 is success; qed");
-
-		CurrencyIdMap::<T>::mutate(id, |maybe_erc20_info| -> DispatchResult {
-			if let Some(erc20_info) = maybe_erc20_info.as_mut() {
-				ensure!(erc20_info.address == address, Error::<T>::CurrencyIdExisted);
-			} else {
-				let info = Erc20Info {
-					address,
-					name: T::EVMBridge::name(InvokeContext {
-						contract: address,
-						sender: Default::default(),
-						origin: Default::default(),
-					})?,
-					symbol: T::EVMBridge::symbol(InvokeContext {
-						contract: address,
-						sender: Default::default(),
-						origin: Default::default(),
-					})?,
-					decimals: T::EVMBridge::decimals(InvokeContext {
-						contract: address,
-						sender: Default::default(),
-						origin: Default::default(),
-					})?,
-				};
-
-				*maybe_erc20_info = Some(info);
-			}
-			Ok(())
-		})
+		CurrencyIdMap::<T>::mutate(
+			Into::<u32>::into(DexShare::Erc20(address)),
+			|maybe_erc20_info| -> DispatchResult {
+				if let Some(erc20_info) = maybe_erc20_info.as_mut() {
+					ensure!(erc20_info.address == address, Error::<T>::CurrencyIdExisted);
+				} else {
+					let info = Erc20Info {
+						address,
+						name: T::EVMBridge::name(InvokeContext {
+							contract: address,
+							sender: Default::default(),
+							origin: Default::default(),
+						})?,
+						symbol: T::EVMBridge::symbol(InvokeContext {
+							contract: address,
+							sender: Default::default(),
+							origin: Default::default(),
+						})?,
+						decimals: T::EVMBridge::decimals(InvokeContext {
+							contract: address,
+							sender: Default::default(),
+							origin: Default::default(),
+						})?,
```

Comment:
Add a constructor fn for `InvokeContext` with `contract` param, and default sender/origin.